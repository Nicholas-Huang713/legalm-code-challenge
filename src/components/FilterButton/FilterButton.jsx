import { useState, useEffect } from "react";
const optionsArr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export default function FilterButton({ setFilter }) {
  const [selectedOption, setSelectedOption] = useState(0);

  const renderOptions = () => {
    return optionsArr.map((val) => (
      <option value={val} key={val}>
        {val}
      </option>
    ));
  };

  const handleChange = (event) => setSelectedOption(event.target.value);

  useEffect(() => {
    setFilter(selectedOption);
  }, [selectedOption]);

  return (
    <>
      <label htmlFor="filter">Filter by Exp </label>
      <select id="filter" value={selectedOption} onChange={handleChange}>
        <option value="none">None</option>
        {renderOptions()}
      </select>
    </>
  );
}
