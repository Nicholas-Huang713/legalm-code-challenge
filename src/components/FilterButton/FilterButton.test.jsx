import { render, screen, fireEvent } from "@testing-library/react";
import FilterButton from "./FilterButton";

describe("FilterButton Component", () => {
  it("renders FilterButton and changes the filter", () => {
    const setFilter = vi.fn();
    const { container } = render(<FilterButton setFilter={setFilter} />);
    expect(screen.getByLabelText(/Filter by Exp/i)).toBeInTheDocument();
    const selectBox = screen.getByLabelText(/Filter by Exp/i);
    fireEvent.change(selectBox, { target: { value: "5" } });
    expect(setFilter).toHaveBeenCalledWith("5");
    fireEvent.change(selectBox, { target: { value: "10" } });
    expect(setFilter).toHaveBeenCalledWith("10");
    expect(container).toMatchSnapshot();
  });
});
