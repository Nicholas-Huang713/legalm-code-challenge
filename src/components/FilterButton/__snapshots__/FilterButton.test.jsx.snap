// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterButton Component > renders FilterButton and changes the filter 1`] = `
<div>
  <label
    for="filter"
  >
    Filter by Exp 
  </label>
  <select
    id="filter"
  >
    <option
      value="none"
    >
      None
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
</div>
`;
